### DISCUSSION TITLE
TX Mode

### DISCUSSION THREAD

**Discussion by Williamreen** [peterantypas/maiana#57] [General]
Hi peter , I think it's a very coolproject,in my own experiment.I can get data from rx mode and I can see gps and Ais data from uart but I can't transfer to any channel.When I checked the TX signal, I didn't see anything. Can you help me

**Comment by peterantypas** [comment]
Are you trying to transmit actual station data or test packets?

For test packets, you need to send the "tx test" command via the UART. This will send a 1 second randomly modulated signal at one of the two channels. Suggest you do that against a measurement instrument, not a live antenna. 

For actual station data you need to provision first via the "station" command and the GPS must have obtained a position fix. Assuming that you're stationary, transmission interval is 2 minutes for position reports, longer for metadata reports. Absent a GPS fix, no transmission will occur.

Read the source for "CommandProcessor.cpp"  to see the full list of CLI commands. There are no docs right now.

**Reply by peterantypas** [reply]
Also, depending on which board you built, you may want to check the status of the TX_EN signal.

**Comment by Williamreen** [comment]
Thanks for your reply.I work with  board version 10.0.1.When I sent the "tx test" from UART, I was able to observe the signal CH_87 or CH_88.I want to send the station data, I think it will happen automatically.My string is "station 987654321, "NAUT" ,"" ,0,0,0,0,0" as I sent over uart, I think I'm missing somewhere .I can see the test data over the sdrsharp but I can't see the station data.I think it should automatically generate ais data and send it every 2 minutes .at this point I'm stuck.Thanks in advance for your help.

**Reply by peterantypas** [reply]
A couple of things:

- Do not use quotes for any of the strings in the "station" command
- Are you sure your SDR is fast enough to catch a 26ms transmission? Even some spectrum analyzers will have a hard time with that. This is the exact reason why I made test packets 9600 bits long.

**Reply by peterantypas** [reply]
Another thing to check is the HOST_TX_STATUS signal from the board output. If you pull this up to some voltage, can you catch a falling edge with a scope? Then you know transmission happened. Absent that, something else is preventing it.

On this board, HOST_TX_OFF should not be driven or it will prevent transmission.

**Reply by Williamreen** [reply]
Peter, thank you so much for your suggestions .I did what you suggested, and my problem was solved.It's about the SDR.This project is really impressive :)
